File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/adapters/Generics/GenericViewHolder.java
package com.atomykcoder.atomykplay.adapters.Generics;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import com.atomykcoder.atomykplay.interfaces.IBindableViewHolder;

public class GenericViewHolder<T> extends RecyclerView.ViewHolder implements IBindableViewHolder<T> {
    public GenericViewHolder(View itemView) {
        super(itemView);
    }

    @Override
    public void onBind(T item) {
    }
}

File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/adapters/generics/GenericViewHolder.kt
package com.atomykcoder.atomykplay.adapters.generics

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.atomykcoder.atomykplay.interfaces.IBindableViewHolder

open class GenericViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(
    itemView
), IBindableViewHolder<T> {
    override fun onBind(item: T) {}
}